svg.js.d.ts

```diff
@@ -82,58 +105,505 @@ declare module "@svgdotjs/svg.js" {
+        T extends SVGTextPathElement ? TextPath :
+        T extends SVGGElement ? G :
+        T extends SVGLineElement ? Line :
+        T extends SVGPolylineElement ? Polyline :
+        T extends SVGPolygonElement ? Polygon :
+        T extends SVGGradientElement ? Gradient :
+        T extends SVGImageElement ? Image :
+        T extends SVGEllipseElement ? Ellipse :
+        T extends SVGMaskElement ? Mask :
+        T extends SVGMarkerElement ? Marker :
+        T extends SVGClipPathElement ? ClipPath :
+        T extends SVGTSpanElement ? Tspan :
+        T extends SVGSymbolElement ? Symbol :
+        T extends SVGUseElement ? Use : Element
+
+    // element type as string
+    type SvgType = "svg"
+    type ClipPathType = "clipPath"
+    type TextType = "text"
+    type GType = "g"
+    type AType = "a"
+
+    type ParentElement = SvgType | GType | AType
+
+    type AttrTypeMapping<T> =
+        T extends Rect ? RectAttr : GlobalAttr
+
+    type StyleTypeMapping<T> =
+        T extends Element ? SVGStyleDeclaration : CSSStyleDeclaration
+
+    type ElementAlias = Dom | Svg | Rect | Line | Polygon | Polyline | Ellipse | ClipPath | Use |
+        Text | Path | TextPath | Circle | G | Gradient | Image | Element
+
+    type AttributeReference = "href" | "marker-start" | "marker-mid" | "marker-end" | "mask" |
+        "clip-path" | "filter" | "fill"
+
+    // ************* SVG.JS Type Declaration *************
+    // ********** Locate in directory src/types **********
+
+    // SVGArray.js
+    /**
+     * Type alias to native array.
+     *
+     * **Caution**: If argument is a string, generic type must be a number or array of number and
+     * the string is format as a concatenate of number separate by comma.
+     * This is expensive to build runtime type check for such as case so please use it carefully.
+     */
+    type ArrayAlias<T> = BuiltInArray<T> | T[] | string
+
+    class Array<T> extends BuiltInArray<T> {
+        constructor(array?: ArrayAlias<T>);
+
+        /**
+         * Return array of generic T however it's flatten array by 1 level as it using `apply` function.
+         * For example: if T is a `number[]` which is the number of 2 dimension `Array<number[]>` the result will be `number[]`
+         */
+        toArray(): any[]
+        /**
+         * return a concatenated string of each element separated by space
+         */
+        toString(): string
+        valueOf(): T[]
+        clone(): Array<T>
+        toSet(): Set<T>
+        parse(a?: ArrayAlias<T>): T[]
+        to(a: any): Morphable;
+    }
+
+    // point.js
+    class Point {
+        x: number;
+        y: number;
+        constructor();
+        constructor(position: CoordinateXY);
+        constructor(point: Point);
+        constructor(position: { x: number, y: number });
+        constructor(x: number, y?: number);
+        clone(): Point;
+        transform(matrix: Matrix): this;
+        transform0(matrix: Matrix): this;

```

Comment:
Its `transformO` not `transform0` - Its an o not a zero. Its short for "optimized"